mmcls/models/backbones/efficientnet_v2.py

```diff
@@ -0,0 +1,330 @@
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 mid_channels,
+                 kernel_size=3,
+                 stride=1,
+                 se_cfg=None,
+                 with_residual=True,
+                 conv_cfg=None,
+                 norm_cfg=dict(type='BN'),
+                 act_cfg=dict(type='ReLU'),
+                 drop_path_rate=0.,
+                 with_cp=False,
+                 init_cfg=None):
+        super(EdgeResidual, self).__init__(init_cfg=init_cfg)
+        assert stride in [1, 2]
+        self.with_cp = with_cp
+        self.drop_path = DropPath(
+            drop_path_rate) if drop_path_rate > 0 else nn.Identity()
+        self.with_se = se_cfg is not None
+        self.with_residual = (
+                stride == 1 and in_channels == out_channels and with_residual)
+
+        if self.with_se:
+            assert isinstance(se_cfg, dict)
+
+        self.conv1 = ConvModule(
+            in_channels=in_channels,
+            out_channels=mid_channels,
+            kernel_size=kernel_size,
+            stride=1,
+            padding=kernel_size // 2,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg)
+
+        if self.with_se:
+            self.se = SELayer(**se_cfg)
+
+        self.conv2 = ConvModule(
+            in_channels=mid_channels,
+            out_channels=out_channels,
+            kernel_size=1,
+            stride=stride,
+            padding=0,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            act_cfg=None)
+
+    def forward(self, x):
+
+        def _inner_forward(x):
+            out = x
+            out = self.conv1(out)
+
+            if self.with_se:
+                out = self.se(out)
+
+            out = self.conv2(out)
+
+            if self.with_residual:
+                return x + self.drop_path(out)
+            else:
+                return out
+
+        if self.with_cp and x.requires_grad:
+            out = cp.checkpoint(_inner_forward, x)
+        else:
+            out = _inner_forward(x)
+
+        return out
+
+
+class ConvWSkip(BaseModule):
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size,
+                 stride=1,
+                 skip=True,
+                 drop_path_rate=0.,
+                 conv_cfg=dict(type='Conv2dAdaptivePadding'),
+                 norm_cfg=dict(type='BN', eps=1e-3, momentum=0.1),
+                 act_cfg=dict(type='Swish'),
+                 init_cfg=None):
+        super(ConvWSkip, self).__init__(init_cfg=init_cfg)
+        self.conv = ConvModule(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg)
+        self.has_skip = skip and stride == 1 and in_channels == out_channels
```

Comment:
```
self.has_skip = skip 
if self.has_skip:
    assert stride == 1 and in_channels == out_channels, (
                "the stride must be 1 and the in_C must be equal with"
                " out_C, when `skip` is True in `ConvWithSkip` .")
```

I think this assertion will be more friendly, if the user wants to have short-cut but set the wrong parameters will have a reminder, not he set but finally did not turn on.

Reply:
We also use the stride and (in_channels=out_channels) as a control condition, for example，if stride==2, even if skip==True, the module will not have skip. All in all , the module only add a parameter to control skip, but this conv also need too conditions. The module means the skip is need, if skip can, so the module include skip, if skip cannot , the module not include skip.

Reply:
We also use the stride and (in_channels=out_channels) as a control condition, for example，if stride==2, even if skip==True, the module will not have skip. All in all , the module only add a parameter to control skip, but this conv also need two conditions. The module means the skip is need, if skip can, so the module include skip, if skip cannot , the module not include skip.

Reply:
I understand the logic here, if strid==2, there is no skip, but it doesn't make sense. Imagine this.

1. the user wants to get a conv with short-cut
```
a = ConvWithSkip(in_C, out_C)
```

2. The user wants a conv with short-cut but stride=2
```
a = ConvWithSkip(in_C, out_C,stride=2)
```
Here we get a conv without a short-cut, which is not what is expected.

**In fact, it is impossible to create a conv with short-cut and stride=2. It is the user's expectation that is wrong, I think the right thing to do is to tell the user that it is not allowed here, not to hide it and help it get a conv without short-cut.**

Reply:
how about :
```
class ConvWithSkip(ConvModule):
    def __init__(*args, has_skip=True, **args):
        super()._init__(*args, **args)
        # some assert lines
        self.has_skip = has_skip
        if self.has_skip:
            self.inplace = False

    def forward(self,
                x: torch.Tensor,
                activate: bool = True,
                norm: bool = True) -> torch.Tensor:
        short_cut = x
        for layer in self.order:
           if layer == 'conv':
                if self.with_explicit_padding:
                    x = self.padding_layer(x)
                x = self.conv(x)
            elif layer == 'norm' and norm and self.with_norm:
                x = self.norm(x)
            elif layer == 'act' and activate and self.with_activation:
                x = self.activate(x)
        return x if not self.has_skip else x + short_cut
```

Reply:
change a little , add self in __init__ , and dropout forward part , is there something wrong?
```
class EnhancedConvModule(ConvModule):
    """Conv with short-cut and droppath."""
    def __init__(self,*args, has_skip=False, drop_path_rate=0, **args):
        super()._init__(*args, **args)
        # some assert lines
        self.has_skip = has_skip
        if self.has_skip:
            assert stride == 1 and in_channels == out_channels, (
                "the stride must be 1 and the in_C must be equal with"
                " out_C, when `skip` is True in `ConvWithSkip` .")
        self.drop_path = DropPath(drop_path_rate) if drop_path_rate else nn.Identity()
        if self.has_skip and self.inplace :
            self.inplace = False
            warning.warn("if there has short-cut in `EnhanceConv`, in place would be False.")

    def forward(self,
                x: torch.Tensor,
                activate: bool = True,
                norm: bool = True) -> torch.Tensor:
        short_cut = x
        for layer in self.order:
           if layer == 'conv':
                if self.with_explicit_padding:
                    x = self.padding_layer(x)
                x = self.conv(x)
            elif layer == 'norm' and norm and self.with_norm:
                x = self.norm(x)
            elif layer == 'act' and activate and self.with_activation:
                x = self.activate(x)
        if self.has_skip:
            x = self.drop_path(x) + short_cut
        return x 
```